public/js/functions.js

```diff
@@ -276,23 +295,44 @@ function _csv(headers, rows) {
 	return str;
 }
 
+function datenum(v, date1904) {
+	if(date1904) v+=1462;
+	var epoch = Date.parse(v);
+	return (epoch - new Date(Date.UTC(1899, 11, 30))) / (24 * 60 * 60 * 1000);
+}
+
+function Boolean(val, fallback){
+	if(typeof val === 'boolean') return val;
+	if(typeof val === 'number') {
+		if(val % 1 == 0 && val <= 1 && val >= 0){
+			return Boolean(val);
+		}
+	}
+	if(val != undefined){
+		var str = val.toLowerCase();
+		if(str == "true" || str == "1") return true;
+		if(str == "false" || str == "0") return false;
+	}
+	return fallback;
+}
+
 function File(name, f, type, displayName) {
 	var a = document.createElement('a');
 	a.appendChild(document.createTextNode(displayName || name));
-	a.href = '#';
-	a.className = "table-download-link";
-	a.onclick = function() {
-		saveAs(new Blob([f()], {type: type}), name);
+	a.href = '#';	
```

Comment:
Moved data out of the function so that URL only renders after function has evaluated.